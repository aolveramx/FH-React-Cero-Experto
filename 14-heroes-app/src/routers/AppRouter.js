import {
  BrowserRouter as Router,
  Switch,
  Route,
} from 'react-router-dom';

import DashBoardRouter from './DashBoardRouter';
import LoginScreen from '../login/LoginScreen';

const AppRouter = () => {
  return (
    <Router>
      <div>
        <Switch>
          <Route exact path="/login" component={ LoginScreen } />
          <Route path="/" component={ DashBoardRouter } />
        </Switch>
      </div>
    </Router>
  )
}

export default AppRouter
