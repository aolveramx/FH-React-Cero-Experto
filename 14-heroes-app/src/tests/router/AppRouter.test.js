import '@testing-library/jest-dom'
import { mount } from 'enzyme'
import { AuthContext } from '../../auth/AuthContext';
import AppRouter from '../../routers/AppRouter'

describe('Tests in <AppRouter />', () => {

  const contextValue = {
    dispath: jest.fn(),
    user: {
      logged: false,
    }
  }

  test('should render login if isAuthenticated: false', () => {
    
    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <AppRouter />
      </AuthContext.Provider>
    )

    expect(wrapper).toMatchSnapshot()

  });

  test('should render <MarvelScreen /> if isAuthenticated: true', () => {

    const contextValue = {
      dispath: jest.fn(),
      user: {
        logged: true,
        name: 'Ari'
      }
    }

    const wrapper = mount(
      <AuthContext.Provider value={ contextValue }>
        <AppRouter />
      </AuthContext.Provider>
    )

    expect( wrapper.find('.navbar').exists()).toBe(true)
    
  });
  
});