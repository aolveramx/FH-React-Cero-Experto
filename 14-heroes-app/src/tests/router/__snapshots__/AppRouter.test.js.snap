// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <AppRouter /> should render login if isAuthenticated: false 1`] = `
<div>
  <div
    className="container .t-5"
  >
    <h1>
      Login Screen
    </h1>
    <hr />
    <button
      className="btn btn-primary"
      onClick={[Function]}
    >
      Login
    </button>
  </div>
</div>
`;
